tests/src/test/scala/cats/tests/MonadRecInstancesTests.scala

```diff
@@ -0,0 +1,37 @@
+package cats
+package tests
+
+import cats.data.{OptionT, Xor, XorT}
+
+class MonadRecInstancesTests extends CatsSuite {
+  def tailRecMStackSafety[M[_]](implicit M: MonadRec[M], Eq: Eq[M[Int]]): Unit = {
+    val n = 50000
```

Comment:
Can this test be added as a `MonadRecLaws` and `MonadRecTests` in the `cats-laws` project, with tests for instances being put in `OptionTests`, `XorTests`, etc. ?